sot_hpp/ros_interface.py

```diff
@@ -99,23 +99,29 @@ def clearQueues(self, req):
 
     def readQueue(self, req):
         if self.supervisor is not None:
-            self.supervisor.readQueue(req.data)
+            SoTTime = self.supervisor.sotrobot.device.control.time
+            self.supervisor.readQueue( SoTTime + req.data)
         else:
-            cmd = "supervisor.readQueue(" + str(req.data) + ")"
+            cmd = "supervisor.sotrobot.device.control.time"
+            answer = self.runCommand (cmd)
+            print cmd
+            print answer
+            SoTTime = int(answer.result)
+            cmd = "supervisor.readQueue(" + str(SoTTime + req.data) + ")"
```

Comment:
Sorry, I think I was not clear. There is still two ROS calls here as `runCommand` uses ROS services. I meant that it should be done in class Supervisor.

Reply:
Oh, I forgot that `runCommand` used ROS services…
[4290b80](https://github.com/jmirabel/sot-hpp/pull/1/commits/4290b807de34cef751d0adc36d92cca9c7761961) moves the functionality to the class Supervisor and removes the extra call to `runCommand`.